src/irmin-pack/inode.ml

```diff
@@ -759,6 +759,117 @@ struct
       let hash t = I.hash t.v
       let stable t = I.stable t.v
       let length t = I.length t.v
+
+      module Serde = struct
+        type sexp = Sexplib.Sexp.t
+
+        (* type ptr = { target_hash : hash Lazy.t; mutable target : t option }
+         * and tree = { depth : int; length : int; entries : ptr option array }
+         * and v = Values of value StepMap.t | Tree of tree
+         * and t = { hash : hash Lazy.t; stable : bool; v : v } *)
+
+        let parse_from_string s =
+          let lexbuf = Lexing.from_string s in
+          Sexplib.Sexp.scan_sexp lexbuf
+
+        let parse_from_file f = Sexplib.Sexp.load_sexp f
+        let sexp_of_hash h = Sexplib.Sexp.(Atom (Fmt.str "%a" pp_hash h))
+        let sexp_of_lazy_hash h = Sexplib.Std.sexp_of_lazy_t sexp_of_hash h
+
+        let sexp_of_metadata m =
+          Sexplib.Sexp.Atom (Fmt.str "%a" (Irmin.Type.pp metadata_t) m)
+
+        (* A kind is either a Content (B for blob) or a Node (N) *)
+        let sexp_of_kind k =
+          let open Sexplib.Sexp in
+          match k with
+          | `Contents (h, m) ->
+              List [ Atom "B"; sexp_of_hash h; sexp_of_metadata m ]
+          | `Node h -> List [ Atom "N"; sexp_of_hash h ]
```

Comment:
From the standpoint of someone who reads this serialisation from a file, what will they want to do with these 2 hashes (computed using `sexp_of_hash h`) ?

Should we also include the values corresponding to these hashes in the file?

Reply:
Including the value would mean fetching them from the pack, right? I'm actually not sure why I put these hashes there but I guess you don't want to have the full value in it if you know you can retrieve it from the hash given.

Reply:
`Including the value would mean fetching them from the pack, right?` yes, and the serialisation shouldn't do more than storing the hashes here, your code is fine. 

I'm wondering about the files we're going to produce. I suppose we are going to need to include the values there

Reply:
Since this is just a serialisation for inodes and not the blobs I don't think including the values is of any importance but I may be wrong :-)